data/posts.js

```diff
@@ -0,0 +1,141 @@
+            tags: tags,
+            postTime: JSON.stringify(new Date()),
+            rating: 0,
+            resolvedStatus: false,
+            commentIds: []
+        };
+        const newInsertInformation = await postCollection.insertOne(newPost);
+        if (newInsertInformation.insertedCount === 0) throw 'Insert failed!';
+        const newID = newInsertInformation.insertedId.toString();
+        return await this.getPostById(newID);
+    },
+
+    async getAllPosts() {
+        const postCollection = await posts();
+        const postList = await postCollection.find({}).toArray();
+        if (!postList) throw 'No posts in system!';
+        postList.forEach((post) => {
+            post._id = post._id.toString();
+        });
+        return postList;
+    },
+    async getPostById(id) {
+        let postId = idCheck(id);
+        const postCollection = await posts();
+        const post = await postCollection.findOne({ _postId: postId });
+        if (!post) throw 'post not found';
+        post._postId = post._postId.toString();
+        return post;
+    },
+    // user edits a post
+    async updatePost(id, updatedPost) {
+        // title, postContent, tags, rating, resolvedStatus can be updated
+        // postTime will be updated as well
+        let postId = idCheck(id);
+        let postUpdateInfo = {};
+        if (updatedPost.title) {
+            if (!stringCheck(updatedPost.title))
+                throw 'title attribute must be a nonempty string';
+            postUpdateInfo.title = updatedPost.title;
+        }
+        if (updatedPost.postContent) {
+            if (!stringCheck(updatedPost.postContent))
+                throw 'postContent attribute must be a nonempty string';
+            postUpdateInfo.postContent = updatedPost.postContent;
+        }
+        if (updatedPost.tags) {
+            if (
+                !(
+                    Array.isArray(updatedPost.tags) &&
+                    updatedPost.tags.every(stringCheck)
+                )
+            )
+                throw 'tags attribute must be an array of strings';
+            postUpdateInfo.tags = updatedPost.tags;
+        }
+        if (updatedPost.rating) {
+            if (!(stringCheck(updatedPost.rating) && typeof rating == 'number'))
+                throw 'rating attribute must be a nonempty string number';
+            postUpdateInfo.rating = updatedPost.rating;
+        }
+        if (updatedPost.resolvedStatus) {
+            if (typeof updatedPost.resolvedStatus !== 'boolean')
+                throw 'resolvedStatus attribute must be a boolean';
+            postUpdateInfo.resolvedStatus = updatedPost.resolvedStatus;
+        }
+        const postCollection = await posts();
+        const updateInfo = await postCollection.updateOne(
+            { _id: postId, postTime: JSON.stringify(new Date()) },
+            { $set: postUpdateInfo }
+        );
+        if (!updateInfo.matchedCount && !updateInfo.modifiedCount)
```

Comment:
!update.matchedCount is causing an error. 
`if (!updateInfo && !updateInfo.modifiedCount) throw "Update failed";`
This makes it work though.